utils.js

```diff
@@ -952,6 +952,112 @@ const getGoPkgLicense = async function (repoMetadata) {
 };
 exports.getGoPkgLicense = getGoPkgLicense;
 
+const getGoPkgComponent = async function (gosumData, group, name, version) {
+  let pkg = {};
+  if (!gosumData) {
+    return pkgList;
+  }
+  const pkgs = gosumData.split("\n");
+  const query = `${group}/${name} ${version}`;
+  for (let i in pkgs) {
+    const l = pkgs[i];
+
+    if (l.includes(query)) {
+      const tmpA = l.split(" ");
+      const sumVersion = tmpA[1].replace("/go.mod", "");
+      const sumHash = tmpA[tmpA.length - 1].replace("h1:", "sha256-");
+      let license = undefined;
+      if (process.env.FETCH_LICENSE) {
+        if (DEBUG_MODE) {
+          console.log(
+            `About to fetch go package license information for ${group}:${name}`
+          );
+        }
+        license = await getGoPkgLicense({
+          group: group,
+          name: name,
+        });
+      }
+      pkg = {
+        group: group,
+        name: name,
+        version: sumVersion,
+        _integrity: sumHash,
+        license: license,
+      };
+    }
+  }
+  return pkg;
+};
+exports.getGoPkgComponent = getGoPkgComponent;
+
+const parseGoModData = async function (goModData, gosumData) {
+  const pkgComponentsList = [];
+  let isModReplacement  = false;
+
+  if (!goModData) {
+    return pkgComponentsList;
+  }
+
+  const pkgs = goModData.split("\n");
+  for (let i in pkgs) {
+    let l = pkgs[i];
+
+    // Skip go.mod file headers, whitespace, and/or comments
+    if (l.includes("module ") || l.includes("go ") || l.includes(")") || l.trim() === "" || l.trim().startsWith("//")){
+      continue;
+    }
+
+    // Handle required modules separately from replacement modules to ensure accuracy when parsing component data.
+    if (l.includes("require (")) {
+      isModReplacement = false;
+      continue;
+    } else if (l.includes("replace (")) {
+      isModReplacement = true;
+      continue;
+    } else if (l.includes("replace ")) {
+      // If this is an inline replacement, drop the word replace 
+      // (eg; "replace google.golang.org/grpc => google.golang.org/grpc v1.21.0" becomes " google.golang.org/grpc => google.golang.org/grpc v1.21.0")
+      l = l.replace("replace", "");
+      isModReplacement = true;
+    }
+
+    const tmpA = l.trim().split(" ");
+
+    if (!isModReplacement) {
+      // Add group, name and version component properties for required modules
+      let group = path.dirname(tmpA[0]);
+      const name = path.basename(tmpA[0]);
+      if (group === ".") {
+        group = name;
+      }
+      const version = tmpA[1];
+      await getGoPkgComponent(gosumData, group, name, version)
```

Comment:
This seems to be an expensive operation since it is iterating all the lines in the text and constructing the string look to determine the hash. Is it possible to build a dict of key and hash as the value once and refer it here directly?